types.go

```diff
@@ -42,21 +48,52 @@ func (r Reference) ByteSize() int {
 	return len(r)
 }
 
-func toBytes(data interface{}) ([]byte, error) {
-	switch castData := data.(type) {
-	case []uint8:
-		return castData, nil
-	case uint32:
-		var buf [4]byte
-		binary.BigEndian.PutUint32(buf[:], castData)
-		return buf[:], nil
+// Scalar represents a JSON or JSON-LD scalar (string, number, true or false)
+type Scalar struct {
+	value interface{}
+}
+
+// ErrInvalidValue is returned when an invalid value is parsed
+var ErrInvalidValue = errors.New("invalid value")
+
+// ScalarParse returns a Scalar based on an interface value. It returns ErrInvalidValue for unsupported values.
+func ScalarParse(value interface{}) (Scalar, error) {
+	switch value.(type) {
+	case bool:
+		return Scalar{value: value}, nil
+	case string:
+		return Scalar{value: value}, nil
+	case float64:
+		return Scalar{value: value}, nil
+	}
+	// not possible
+	return Scalar{}, ErrInvalidValue
+}
+
+// ScalarMustParse returns a Scalar based on an interface value. It panics when the value is not supported.
+func ScalarMustParse(value interface{}) Scalar {
+	s, err := ScalarParse(value)
+	if err != nil {
+		panic(err)
+	}
+	return s
+}
+
+// Bytes returns the byte representation of the Scalar
+func (s Scalar) Bytes() []byte {
+	switch castData := s.value.(type) {
+	case bool:
+		if castData {
+			return []byte{1}
+		}
+		return []byte{0}
 	case string:
-		return []byte(castData), nil
+		return []byte(castData)
 	case float64:
 		var buf [8]byte
 		binary.BigEndian.PutUint64(buf[:], math.Float64bits(castData))
```

Comment:
didn't see this before, does it work for negative numbers?

Reply:
added test for it